### PR TITLE
[NETBEANS-4284] - cleanup a bunch of raw type warnings..

### PR DESCRIPTION
**Pull request by BradWalker**
Cleanup a bunch of raw type warnings related to the following..

[repeat] /home/bwalker/src/netbeans/platform/openide.util/src/org/openide/util/MapFormat.java:91: warning: [rawtypes] found raw type: Map
[repeat] public MapFormat(Map arguments) {
[repeat] ^
[repeat] missing type arguments for generic class Map<K,V>
[repeat] where K,V are type-variables:
[repeat] K extends Object declared in interface Map
[repeat] V extends Object declared in interface Map
[repeat] /home/bwalker/src/netbeans/platform/openide.util/src/org/openide/util/MapFormat.java:105: warning: [rawtypes]

Also, updated Iterator raw types warnings..

### CODE REVIEW

File: java/maven.grammar/src/org/netbeans/modules/maven/codegen/ExclusionGenerator.java [hunk 1]
```diff
@@ -90,9 +90,9 @@ public int write() {
                                 // now check the active profiles for the dependency..
                                 List<String> profileNames = new ArrayList<String>();
                                 NbMavenProject project = prj.getLookup().lookup(NbMavenProject.class);
-                                Iterator it = project.getMavenProject().getActiveProfiles().iterator();
+                                Iterator<org.apache.maven.model.Profile> it = project.getMavenProject().getActiveProfiles().iterator();
```
Comment (hectorespert) [inline java/maven.grammar/src/org/netbeans/modules/maven/codegen/ExclusionGenerator.java:93]: Why not import this class?

Comment (BradWalker) [reply java/maven.grammar/src/org/netbeans/modules/maven/codegen/ExclusionGenerator.java:93]: Hey @hectorespert , I wish I could do this.. But, there is already another Profile class being used, org.netbeans.modules.maven.model.pom.Profile. And it's not easy to distinguish between the two..

So we're going to have to leave this one alone..

Thanks for the feedback and hopefully you can now approve it for 12.1..

File: ide/projectuiapi/src/org/netbeans/spi/project/ui/support/NodeFactorySupport.java [hunk 2]
```diff
@@ -174,7 +174,7 @@ public void run() {
                             NodeList<?> lst = factory.createNodes(project);
                             assert lst != null : "Factory " + factory.getClass() + " has broken the NodeFactory contract."; //NOI18N
                             lst.addNotify();
-                            List<?> objects = lst.keys();
+                            List objects = lst.keys();
```
Comment (matthiasblaesing) [inline ide/projectuiapi/src/org/netbeans/spi/project/ui/support/NodeFactorySupport.java:177]: Is the switch to raw types here intented? This looks strange given the target of this PR.

**Review by matthiasblaesing — COMMENTED** [review]
Looks sane in general. I have left one inline comment.

File: ide/projectuiapi/src/org/netbeans/spi/project/ui/support/NodeFactorySupport.java (hunk 2 above)
Comment (BradWalker) [reply ide/projectuiapi/src/org/netbeans/spi/project/ui/support/NodeFactorySupport.java]: I've reverted the file NodeFactorySupport.java to the original.. It was too problematic to fix.. Will clean up later as there's plenty to do..

**Review by matthiasblaesing — COMMENTED** [review]
I found another small problem and this PR holds the testng update commit, this seems like the rebase was not completely correct.

File: ide/xml/src/org/netbeans/modules/xml/actions/CollectSystemAction.java [hunk 3]
```diff
@@ -44,7 +44,7 @@
     /** empty array of menu items */
     static JMenuItem[] NONE = new JMenuItem[] {};
 
-    protected final List registeredAction = new ArrayList();
+    protected final List<Object> registeredAction = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (BradWalker) [reply ide/xml/src/org/netbeans/modules/xml/actions/CollectSystemAction.java]: Hey @matthiasblaesing , fixed..

**Comment by matthiasblaesing** [comment]
Thank you.

**Pull request merged**